runtime/lua/vim/keymap.lua

```diff
@@ -135,4 +135,112 @@ function keymap.del(modes, lhs, opts)
   end
 end
 
+--- @class vim.keymap.get.Opts
+--- @inlinedoc
+---
+--- Lhs of mapping
+--- @field lhs? string
+---
+--- Patter to match against rhs of mapping
+--- @field rhs? string
+---
+--- Get a mapping for a certain buffer
+--- @field buffer? integer|boolean
+
+--- @class vim.keymap.get.Return.Opts
+--- @inlinedoc
+---
+--- If they mapping is an expr mapping
+--- @field expr boolean
+---
+--- If they mapping is a buffer-local mapping
+--- @field buffer boolean
+---
+--- Description of the mapping
+--- @field desc? string
+
+--- @class vim.keymap.get.Return
```

Comment:
Can/should `keymap.set` accept an overload so that this works:

    vim.keymap.set(vim.keymap.get(...))